test/lib/pine-in-process.ts

```diff
@@ -39,6 +40,10 @@ export async function forkInit() {
 		if (process.send) {
 			process.send({ init: 'success' });
 		}
+
+		process.on('SIGUSR1', () => {
```

Comment:
I grepped around in balena-api/open-balena-api and didn't find any other references to the `SIGUSR1` signal. I'm sure you have a good reason for choosing this signal, so could let me know and also add a comment about the reasoning?

Reply:
So the long story is that I wanted to use SIGTERM to be as similar as possible, however, handling sigterm makes the tests break as if I don't do `process.exit()` on it, the tests just hang after running.

Then I had the idea of using a custom signal (such as sigusr1 or sigusr2). This works, however, it sucks as it is a bit too magical and messes up with nodejs default usage of sigusr signals for debugging see https://nodejs.org/api/process.html#signal-events

My current approach now detatches from using any default unix signal to a list of custom pine_test signals handled using the standard nodejs `process.on('message')`. WDYT?

Reply:
Sounds good to me. At the end of the day, we just want need to prove that the new `stop()` does what it should. Maybe add a comment at the message handler in `pine-in-process.ts` to explain that this is an example of how a consumer could be configured to call `worker.stop()`.